ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLAgentExecutor.java

```diff
@@ -104,20 +105,25 @@ public MLAgentExecutor(
     }
 
     @Override
-    public void execute(Input input, String tenantId, ActionListener<Output> listener) {
+    public void onMultiTenancyEnabledChanged(boolean isEnabled) {
+        this.isMultiTenancyEnabled = isEnabled;
+    }
+
+    @Override
+    public void execute(Input input, ActionListener<Output> listener) {
         if (!(input instanceof AgentMLInput)) {
             throw new IllegalArgumentException("wrong input");
         }
         AgentMLInput agentMLInput = (AgentMLInput) input;
         String agentId = agentMLInput.getAgentId();
-        String agentTenantId = agentMLInput.getTenantId();
+        String tenantId = agentMLInput.getTenantId();
 
         RemoteInferenceInputDataSet inputDataSet = (RemoteInferenceInputDataSet) agentMLInput.getInputDataset();
         if (inputDataSet == null || inputDataSet.getParameters() == null) {
             throw new IllegalArgumentException("Agent input data can not be empty.");
         }
 
-        if (isMultiTenancyEnabled && !Objects.equals(tenantId, agentTenantId)) {
+        if (isMultiTenancyEnabled && tenantId == null) {
```

Comment:
Will `tenantId` of the agent returned as part of `GetDataObjectRequest` same as provided tenantId?

Reply:
This check only cares about, when multi-tenancy is enabled, is tenantId is provided in the agentMLInput or not.

Then given tenant id vs actual tenant id of the agent check is in line 175